### ISSUE TITLE
Using expressions in pivot queries

### ISSUE THREAD

**Issue by Jigsaw5279** [laravel/framework#50787]
### Laravel Version

10.48.3

### PHP Version

8.3.3

### Database Driver & Version

mysqlnd 8.3.3

### Description

This issue is similar to #49894, but for querying pivot relations with raw expressions.

In Laravel 9 it was possible to do this:

```php
$item->someMorphToMany()
     ->wherePivotNotIn(
                DB::raw("CONCAT(some_field, '_', other_field)"),
                ['a_b', 'c_d']
            )->detach();
```

which would produce a where statement like `CONCAT(some_field, '_', other_field) not in (?, ?)` which is to be expected.

With the changes of enforcing manual casting of the expression to its raw value, I can no longer get `wherePivotNotIn` to produce the expected output.

```php
$item->someMorphToMany()
     ->wherePivotNotIn(
                DB::raw("CONCAT(some_field, '_', other_field)")->getValue(DB::connection()->getQueryGrammar()),
                ['a_b', 'c_d']
            )->detach();
```

now produces the following where clause

```sql
`database`.`pivot_table`.`CONCAT(some_field, '_', other_field)` not in (?, ?)`
```

which is correctly identified by mysql as an unknown field.

Following the callgraph, I can get rid of the database and table prefix, by prefixing the concatenated fields myself (see https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php#L1527 ).

But then somehwhere in the wrap function it gets mangled again:
```php
$item->someMorphToMany()
     ->wherePivotNotIn(
                DB::raw("CONCAT(table.some_field, '_', table.other_field)")->getValue(DB::connection()->getQueryGrammar()),
                ['a_b', 'c_d']
            )->detach();
```

```sql
`CONCAT(table`.`some_field, '_', table`.`other_field)` not in (?, ?)
```

I'm not confident enough to fiddle around in the guts of eloquent to create a PR for this, so any help would be greatly appreciated

### Steps To Reproduce

1. Setup a `morphToMany` relationship
2. Try to query the relationship with a raw expression constraint to the pivot

**Comment by driesvints** [comment]
@tpetry is this something we need to fix as well?

**Comment by tpetry** [comment]
Yeah, special handling in the core is required to unwrap expressions in this case. But due to personal stuff, I cant work on this currently.

**Comment by driesvints** [comment]
No worries at all @tpetry. Will leave this one open for anyone to pick up.

**Comment by plumthedev** [comment]
Thanks @Jigsaw5279 for reporting that, I have fixed that in #50849 
However, I see that your issue is related to Laravel 10.x and I made MR to 11.x according to [Which branch](https://laravel.com/docs/11.x/contributions#which-branch) - is it okay @driesvints?
